lib/mysql.js

```diff
@@ -550,25 +550,48 @@ MySQL.prototype.ping = function(cb) {
 
 MySQL.prototype.buildExpression = function(columnName, operator, operatorValue,
   propertyDefinition) {
-  if (operator === 'regexp') {
-    let clause = columnName + ' REGEXP ?';
-    // By default, MySQL regexp is not case sensitive. (https://dev.mysql.com/doc/refman/5.7/en/regexp.html)
-    // To allow case sensitive regexp query, it has to be binded to a `BINARY` type.
-    // If ignore case is not specified, search it as case sensitive.
-    if (!operatorValue.ignoreCase) {
-      clause = columnName + ' REGEXP BINARY ?';
-    }
-
-    if (operatorValue.ignoreCase)
-      g.warn('{{MySQL}} {{regex}} syntax does not respect the {{`i`}} flag');
-    if (operatorValue.global)
-      g.warn('{{MySQL}} {{regex}} syntax does not respect the {{`g`}} flag');
+  let clause;
+  switch (operator) {
+    case 'regexp':
+      clause = columnName + ' REGEXP ?';
+      // By default, MySQL regexp is not case sensitive. (https://dev.mysql.com/doc/refman/5.7/en/regexp.html)
+      // To allow case sensitive regexp query, it has to be binded to a `BINARY` type.
+      // If ignore case is not specified, search it as case sensitive.
+      if (!operatorValue.ignoreCase) {
+        clause = columnName + ' REGEXP BINARY ?';
+      }
 
-    if (operatorValue.multiline)
-      g.warn('{{MySQL}} {{regex}} syntax does not respect the {{`m`}} flag');
+      if (operatorValue.ignoreCase)
+        g.warn('{{MySQL}} {{regex}} syntax does not respect the {{`i`}} flag');
+      if (operatorValue.global)
+        g.warn('{{MySQL}} {{regex}} syntax does not respect the {{`g`}} flag');
+
+      if (operatorValue.multiline)
+        g.warn('{{MySQL}} {{regex}} syntax does not respect the {{`m`}} flag');
+
+      return new ParameterizedSQL(clause,
+        [operatorValue.source]);
+    case 'nlmatch':
+    case 'qematch':
+    case 'boolmatch':
+    case 'match':
+      let mode;
+      switch (operator) {
+        case 'boolmatch':
+          mode = ' IN BOOLEAN MODE';
+          break;
+        case 'nlmatch':
+          mode = ' IN NATURAL LANGUAGE MODE';
+          break;
+        case 'qematch':
+          mode = ' WITH QUERY EXPANSION';
+          break;
+        default:
+          mode = '';
```

Comment:
Reading from the docs https://dev.mysql.com/doc/refman/8.0/en/fulltext-search.html, there has to be some expressions after `AGAINST`. I wonder if we should make one of the mode values as default instead of an empty string.

Reply:
If no modifier is provided after `AGAINST`, the `Natural Language` mode is used by default as mentioned [here](https://dev.mysql.com/doc/refman/8.0/en/fulltext-natural-language.html) - "**By default** or with the IN NATURAL LANGUAGE MODE modifier, the MATCH() function performs a natural language search for a string against a text collection".

Reply:
@dhmlau , that's fine, because it is this statement `   return new ParameterizedSQL(clause, [operatorValue]); ` the one that will add the expression you are talking about, it is resolved as the `?` inside the `ParameterizedSQL` later.